lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
Totally inspired by multiple independent user requests to be able to manipulate the field before it is written out to disk for both `PP` and `GRIB`.

At this point, I'm happy to argue the case for saver callback support for only `PP` and `GRIB`, and exclude `NetCDF` ... if that makes a difference.